kedro_mlflow/io/mlflow_metrics_dataset.py

```diff
@@ -0,0 +1,95 @@
+import json
+import os
+from pathlib import Path, PurePosixPath
+from typing import Any, Dict, List, Optional, Union
+
+import mlflow
+from cachetools import Cache
+from kedro.io import AbstractVersionedDataSet
+from kedro.io.core import DataSetError, Version
+
+
+class MlflowMetricsDataSet(AbstractVersionedDataSet):
+    """This class represent mlflow metrics.
+
+    It decorates their ``save`` method to log the metrics in mlflow when
+    ``save`` is called.
+    """
+
+    DEFAULT_SAVE_ARGS: Dict[str, Any] = {"indent": 2}
+
+    def __init__(
+        self,
+        prefix: Optional[str] = None,
+        filepath: Optional[Union[str, PurePosixPath]] = None,
+        version: Optional[Version] = None,
+    ):
+        """Initialise MLflowMetricsDataSet.
+
+        Args:
+            prefix (Optional[str]): Prefix for metrics logged in mlflow.
+            filepath (Optional[Union[str, Path]]): Path to file where dataset
+                will be saved. If not set metrics will be log in mlflow and
+                will not be saved on disk.
+            version (Optional[Version]): If specified, should be an instance of
+                ``kedro.io.core.Version``. If its ``load`` attribute is
+                None, the latest version will be loaded. If its ``save``
+                attribute is None, save version will be autogenerated.
+        """
+        self._filepath = PurePosixPath(filepath) if filepath else filepath
+        self.prefix = prefix
```

Comment:
I would suggest storing `prefix` as `_prefix` to keep it consistent with how kedro is assigning everything to `self` in `__init__`.